metrics_api/test/opentelemetry/opentelemetry_test.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+describe OpenTelemetry do
+  after do
+    # TODO: After Metrics SDK is incoporated into OpenTelemetry SDK, move this
+    # to OpenTelemetry::TestHelpers.reset_opentelemetry
+    OpenTelemetry.instance_variable_set(
+      :@meter_provider,
+      OpenTelemetry::Internal::ProxyMeterProvider.new
+    )
+  end
+
+  describe '#meter_provider and #meter_provider=' do
+    it 'initializes with a global instance of ProxyMeterProvider' do
+      assert(OpenTelemetry.meter_provider.is_a?(OpenTelemetry::Internal::ProxyMeterProvider))
+    end
```

Comment:
I'd like to exercise that `OpenTelemetry.meter_provider` initially returns a `ProxyMeterProvider`. This test kind of exercises that, but it is not ideal. If this test runs first, then it exercises that; otherwise, it doesn't: it just exercises the reset

I was thinking if a proper reset would be calling `load 'opentelemetry-metrics-api.rb` to reload the file instead of manually setting `@meter_provider` with `instance_variable_set` like `OpenTelemetry::TestHelpers` suggests

Reply:
Yeah it's a bit tricky if you're looking to test the initial state before any tests run. I didn't put too much importance on this because in reality all you're testing with that is that [this line of code](https://github.com/open-telemetry/opentelemetry-ruby/blob/3564977bf652e0977796b3f271b0c10728e9f414/metrics_api/lib/opentelemetry-metrics-api.rb#L20) contains no syntax errors.

>If this test runs first, then it exercises that; otherwise, it doesn't: it just exercises the reset

If there was an issue in this code path, it would show up as a flakey test, which is not nothing.